web/src/builder.rs

```diff
@@ -340,53 +341,119 @@ impl RuffleInstanceBuilder {
+                                }
+                                swf::Tag::DefineFont2(font) => {
+                                    tracing::debug!(
+                                        "Loaded font {} from font swf {font_name}",
+                                        font.name.to_str_lossy(encoding)
                                     );
+                                    player.register_device_font(FontDefinition::SwfTag(
+                                        *font, encoding,
+                                    ));
+                                }
+                                swf::Tag::DefineFont4(font) => {
+                                    let name = font.name.to_str_lossy(encoding);
+                                    if let Some(data) = font.data {
+                                        tracing::debug!(
+                                            "Loaded font {name} from font swf {font_name}"
+                                        );
+                                        player.register_device_font(FontDefinition::FontFile {
+                                            name: name.to_string(),
+                                            is_bold: font.is_bold,
+                                            is_italic: font.is_italic,
+                                            data: data.to_vec(),
+                                            index: 0,
+                                        })
+                                    } else {
+                                        tracing::warn!(
+                                            "Font {name} from font swf {font_name} contains no data"
+                                        );
+                                    }
                                 }
+                                _ => {}
                             }
-                            _ => {}
                         }
+                        continue;
                     }
-                    continue;
                 }
+                tracing::warn!("Font source {font_name} was not recognised (not a valid SWF?)");
             }
-            tracing::warn!("Font source {font_name} was not recognised (not a valid SWF?)");
         }
 
         for (default, names) in &self.default_fonts {
             player.set_default_font(*default, names.clone());
         }
     }
 
+    #[inline]
+    fn register_ttf_face_by_name(
+        url: &String,
+        bytes: Vec<u8>,
+        face: ttf_parser::Face<'_>,
+        number_of_fonts: u32,
+        index: u32,
+        player: &mut Player,
+    ) {
+        let full_name = face
+            .names()
+            .into_iter()
+            .find(|name| name.name_id == ttf_parser::name_id::FULL_NAME)
+            .and_then(|name| name.to_string());
+
+        let name = if let Some(full_name) = full_name {
+            full_name
+        } else if number_of_fonts > 1 {
+            format!("{} {}", url, index + 1)
```

Comment:
How likely is this every going to work? Like how often are SWFs using "Font 1", "Font 2" etc? I think this doesn't happen in practice. It would be simpler to just skip (and warn) unnamed fonts.

Reply:
Not sure if there are unnamed fonts just wanted to handle all cases

Reply:
Can more than one font be registered with the same name in ruffle?

Reply:
So maybe better to keep this logic, it's not hard imo, just a but ugly

Reply:
No. We should not accept obvious useless stuff.

Reply:
Ok I will remove, should I just warn when there is no full name and use the url or fail?

Reply:
I think a warning containing the URL and font index would be fine. (That's what we do for some other failures already)

Reply:
Done